tests/buffering.rs

```diff
@@ -67,12 +67,12 @@ fn test_buffering() {
         .send_all(input_events_stream);
     let _ = rt.block_on(send).unwrap();
 
-    // A race caused by `rt.block_on(send).unwrap()` is handled here. For some
-    // reason, at times less events than were sent actually arrive to the
-    // `source`.
-    // We mitigate that by waiting on the event counter provided by our source
-    // mock.
-    test_util::wait_for_atomic_usize(source_event_counter, |x| x == num_events);
+    // There was a race caused by a channel in the mock source, and this
+    // check is here to ensure it's really gone.
+    test_util::wait_for_atomic_usize(source_event_counter, |x| {
```

Comment:
`test_util::wait_for_atomic_usize` would hide the problem if it comes back again. I meant we for us to do a simple comparison instead of cycled wait. I assume that's supposed to always pass now, right?

Reply:
```rust
assert_eq!(source_event_counter.load(Ordering::Relaxed), num_events);
```